validator_client/src/main.rs

```diff
@@ -119,13 +91,20 @@ fn main() {
     let poll_interval_millis = spec.seconds_per_slot * 1000 / 10; // 10% epoch time precision.
     info!(log, "Starting block producer service"; "polls_per_epoch" => spec.seconds_per_slot * 1000 / poll_interval_millis);
 
+    let keypairs = config
+        .fetch_keys()
+        .expect("Encountered an error while fetching saved keys.")
+        .unwrap_or_else(|| {
```

Comment:
This means that if `fetch_keys` returns `None` it gives an error. This means that `fetch_keys` should just return `Result<>` not `Result<Option<>>` because a `None` seems to be an error anyway. It will simplify the function.